From 2cad6a8a4c31175578943f087e1dbef9f52e6ec3 Mon Sep 17 00:00:00 2001
From: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Date: Fri, 26 Oct 2012 23:15:05 +0800
Subject: [PATCH 252/951] GPIO: clps711x: use platform_device_unregister in
 gpio_clps711x_init()

platform_device_unregister() only calls platform_device_del() and
platform_device_put(), thus use platform_device_unregister() to
simplify the code.

Also the documents in platform.c shows that platform_device_del
and platform_device_put must _only_ be externally called in error
cases.  All other usage is a bug.

dpatch engine is used to auto generate this patch.
(https://github.com/weiyj/dpatch)

Signed-off-by: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Signed-off-by: Linus Walleij <linus.walleij@linaro.org>
---
 drivers/gpio/gpio-clps711x.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/gpio/gpio-clps711x.c b/drivers/gpio/gpio-clps711x.c
index ad181db..ce63b75 100644
--- a/drivers/gpio/gpio-clps711x.c
+++ b/drivers/gpio/gpio-clps711x.c
@@ -162,8 +162,7 @@ static int __init gpio_clps711x_init(void)
 			    GFP_KERNEL);
 	if (!gpio) {
 		dev_err(&pdev->dev, "GPIO allocating memory error\n");
-		platform_device_del(pdev);
-		platform_device_put(pdev);
+		platform_device_unregister(pdev);
 		return -ENOMEM;
 	}
 
-- 
1.8.0

